File: src/core/entities/user.entity.ts
import { BaseEntity } from "./base.entity";
import {
  Column,
  Entity,
  JoinColumn,
  JoinTable,
  ManyToMany,
  OneToMany,
} from "typeorm";
import { DEFAULT_AVATAR_PATH } from "@constants/default.constant";
import { ChannelEntity } from "@entities/channel.entity";
import { MessageEntity } from "@entities/message.entity";

@Entity({ name: "users" })
export class UserEntity extends BaseEntity {
  @Column({ unique: true })
  public email: string;

  @Column({ select: false })
  public password: string;

  @Column({ nullable: true, default: "" })
  public name: string;

  @Column({ nullable: true, default: "" })
  public surname: string;

  @Column({ name: "middle_name", nullable: false, default: "" })
  public middleName: string;

  @Column({
    name: "avatar_path",
    nullable: false,
    default: DEFAULT_AVATAR_PATH,
  })
  public avatarPath: string;

  @ManyToMany(() => ChannelEntity, (channel) => channel.members, { onDelete: "CASCADE" })
  public channels: ChannelEntity[];

  @OneToMany(() => MessageEntity, (message) => message.channel, {onDelete: "CASCADE",})
  public messages: MessageEntity[];
}

File: SorokChatServer.Logic/Entities/UserEntity.cs
using SorokChatServer.Logic.Models;
using SorokChatServer.Postgres.Entities;

namespace SorokChatServer.Logic.Entities;

public class UserEntity : BaseEntity
{
    public UserEntity()
    {
    }

    public UserEntity(long id, DateTime createdAt, DateTime updatedAt, Name name, Email email, HashedPassword password,
        string publicRsaKey)
    {
        Id = id;
        CreatedAt = createdAt;
        UpdatedAt = updatedAt;
        Name = name;
        Email = email;
        Password = password;
        PublicRsaKey = publicRsaKey;
    }

    public Name Name { get; set; }

    public Email Email { get; set; }

    public HashedPassword Password { get; set; }

    public string PublicRsaKey { get; set; }
}